recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -461,8 +461,12 @@ def main(argv):
     else:
         updater = ManagedFirmwareUpdate(args.firmware)
 
-    envlist = updater.get_preserved_uboot_env(args.preserve_list)
+    envlist = None
+    if not args.force:
+        envlist = updater.get_preserved_uboot_env(args.preserve_list)
```

Comment:
While should "-f" not preserve the env if all needed information is provided?

Reply:
From my perspective , '-f' would ignore all the checking.so would not keep the env.
It's better to keep the env through our Managed Firmware Update method.

Reply:
Why is that better? This change makes the interface inconsistent.

Reply:
Forced update itself is a higher risk operation. Based on this, providing more interfaces or options, I think the risk will be higher

It is a safer way to keep environment variables in the managed way of updating firmware

The forced update function should be simple

Reply:
Hi jan. 
if we'd like to keep the environment of the target board, so  we'd like to have the builtin environment (u-boot-initial-env) of the target firmware. But currently we plan to integrate this file into our upgrade tar package,this would public to forum. But neither  public u-boot-initial-env  file nor can find the file in our build/tmp/deploy folder.
so for the "-f" option, current what we do is to erase all environment.